### ISSUE TITLE
Service crashing with 'TypeError: Cannot read property 'charCodeAt' of undefined'

### ISSUE THREAD

**Issue by jtsom** [angular/vscode-ng-language-service#61]
Launch VSCode with cursor in an HTML file, and get a crash. Ending with "The Angular Language Service server crashed 5 times in the last 3 minutes.The server will not be restarted."
```
Search path: /home/john/code/mindstorm/src/app/competition-list
Config file name: /home/john/code/mindstorm/tsconfig.json
Angular Language Service: 4.1.0-beta.0-309bae0
TypeScript: 2.1.5
getTemplateRefrences: 2716ms
Add recursive watcher for: /home/john/code/mindstorm
Opened configuration file /home/john/code/mindstorm/tsconfig.json
getTemplateRefrences: 0ms
/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2401
        if (path.charCodeAt(0) === 47 /* slash */) {
                ^

TypeError: Cannot read property 'charCodeAt' of undefined
    at getRootLength (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2401:17)
    at isRootedDiskPath (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2528:16)
    at Object.toPath (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:1273:36)
    at Object.getSourceFile (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:62995:43)
    at findClassSymbolInContext (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:48529:38)
    at TypeScriptSymbolQuery.getTypeSymbol (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:48023:26)
    at _loop_1 (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45694:44)
    at processReferences (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45705:13)
    at class_1.visitEmbeddedTemplate (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45715:13)
    at EmbeddedTemplateAst.visit (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:16483:24)
[Info  - 10:46:42 AM] Connection to server got closed. Server will restart.
Search path: /home/john/code/mindstorm/src/app/competition-list
Config file name: /home/john/code/mindstorm/tsconfig.json
Angular Language Service: 4.1.0-beta.0-309bae0
TypeScript: 2.1.5
getTemplateRefrences: 2484ms
Add recursive watcher for: /home/john/code/mindstorm
Opened configuration file /home/john/code/mindstorm/tsconfig.json
getTemplateRefrences: 0ms
/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2401
        if (path.charCodeAt(0) === 47 /* slash */) {
                ^

TypeError: Cannot read property 'charCodeAt' of undefined
    at getRootLength (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2401:17)
    at isRootedDiskPath (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:2528:16)
    at Object.toPath (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:1273:36)
    at Object.getSourceFile (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/typescript/lib/typescript.js:62995:43)
    at findClassSymbolInContext (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:48529:38)
    at TypeScriptSymbolQuery.getTypeSymbol (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:48023:26)
    at _loop_1 (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45694:44)
    at processReferences (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45705:13)
    at class_1.visitEmbeddedTemplate (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:45715:13)
    at EmbeddedTemplateAst.visit (/home/john/.vscode-insiders/extensions/Angular.ng-template-0.1.1/server/node_modules/@angular/language-service/bundles/language-service.umd.js:16483:24)
[Info  - 10:46:50 AM] Connection to server got closed. Server will restart.
```

Versions:
```
ng version
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.0.0
node: 7.8.0
os: linux x64
@angular/common: 4.0.1
@angular/compiler: 4.0.1
@angular/compiler-cli: 4.0.1
@angular/core: 4.0.1
@angular/forms: 4.0.1
@angular/http: 4.0.1
@angular/material: 2.0.0-beta.2
@angular/platform-browser: 4.0.1
@angular/platform-browser-dynamic: 4.0.1
@angular/router: 4.0.1
@angular/cli: 1.0.0
```

**Comment by chuckjaz** [comment]
Do you have a project I can clone that reproduces this?

**Comment by jtsom** [comment]
Unfortunately, this is just a small sample app. I opened another app and it didn't have a problem. The .html page that throws the error consists of:

```
<div class="comp-list">
    <div *ngIf="comps | async; let comps; else loading">
        <md-card class="primary" *ngFor="let comp of comps">
            <md-card-title>{{comp.full_name}}</md-card-title>
            <md-card-content>{{comp.contact_name}}</md-card-content>
        </md-card>
    </div>

    <ng-template #loading>Loading comps...</ng-template>
</div>

```

Having the cursor anywhere in the text causes the error.

**Comment by chuckjaz** [comment]
Can you attach a zip file of the project? There is something different about the project, not the template.

**Comment by jtsom** [comment]
Sure.
[test.zip](https://github.com/angular/vscode-ng-language-service/files/893925/test.zip)

**Comment by chuckjaz** [comment]
Thanks! That reproduces it.

**Comment by jtsom** [comment]
I see you've found the issue back in the Angular language service.. cool... I tried updating my sample to remove the ngIf and else div, but it still was crashing. Once I removed the ng-template it stopped crashing. FWIW.

**Comment by chuckjaz** [comment]
It is the template reference that was the issue. I am working to get a release out with this fix (once it lands in master).

**Comment by chuckjaz** [comment]
This is fixed in `0.1.2`
